src/pages/Cart.js

```diff
@@ -0,0 +1,127 @@
+import { useState, useEffect } from "react";
+import TextInput from "../components/input/TextInput";
+import Item from "../components/Item";
+import styled from "styled-components";
+
+const shopItems = [
+  {
+    id: "1",
+    name: "Decken-Sectionaltor Holz",
+    qu: "qm",
+    price: 180,
+    quantity: 1,
+    inCart: false,
+  },
+  {
+    id: "2",
+    name: "Decken-Sectionaltor Stahl",
+    qu: "qm",
+    price: 120,
+    quantity: 1,
+    inCart: false,
+  },
+  {
+    id: "3",
+    name: "Rundlauftor Holz",
+    qu: "qm",
+    price: 80,
+    quantity: 1,
+    inCart: false,
+  },
+  {
+    id: "4",
+    name: "Rundlauftor Aluminium",
+    qu: "qm",
+    price: 150,
+    quantity: 1,
+    inCart: false,
+  },
+];
+
+export default function Cart() {
+  const [data, setData] = useState(shopItems);
+  const [searchString, setSearchString] = useState("");
+  const [toggleID, setToggleID] = useState("");
+
+  const { search } = require("fast-fuzzy");
+
+  const falseArray = [];
+  const cartArray = [];
+
+  useEffect(() => {
+    data.forEach((item) => {
+      if (item.id === toggleID) {
+        item.inCart = !item.inCart;
+        setToggleID("");
+      }
+    });
```

Comment:
```suggestion
    if (!toggleID) return;

    setData(
      data.map((item) => {
        if (item.id === toggleID) {
          setToggleID("");
          return { ...item, inCart: !item.inCart };
        } else {
          return item;
        }
      })
    );
    }

```
Martin is right in pointing out that React does not "think" that way. Whenever you want to change state, you need to use the respective setState method.